### ISSUE TITLE
Auto build using GitHub Actions

### ISSUE THREAD

**Issue by clach04** [mattrossman/forecaswatch2#66]
GitHub Actions (GHA) are free to use/deploy.

https://github.com/mattrossman/forecaswatch2/pull/50 offers an implementation for this for this project. I have tweaked that GHA and I'm using https://github.com/clach04/pebble-forecaswatch2-fcsw2-fcw2/blob/gha-no-make/.github/workflows/main.yml this lets me auto build on push or via the web interface and a commit hash, I can do this on my phone :-) It's literally a single file (compared with the open PR).

You can also perform auto releases but I've not looked into this.

Flagging this up as a potential way to make building less of chore.

It's how I shared a test build for resolution of https://github.com/mattrossman/forecaswatch2/issues/60

**Comment by mattrossman** [comment]
Tbh I don't mind manually releasing, I don't have to do it often. Building is easy enough with the pebble docker image, the hardest part for me is local dev with a working emulator. On my old laptop I just barely got the pebble SDK running natively. On my new M1 Mac I can't get it running natively and VirtualBox is not yet supported on M1.

For this release I used the [rebble/pebble-sdk](https://hub.docker.com/r/rebble/pebble-sdk) docker image to build, but I had to use my pebble hardware to debug. It would be much easier if I could just use the emulator.

**Comment by clach04** [comment]
Yeah, targeting new CPU in theory is possible but going to be some work.

I'm using either docker and the emu in there across to X11 in the host OS or native (old Ubuntu). See https://github.com/clach04/docker-pebble-dev/wiki for notes on using the emulator with Docker, it works really well. I previously used VirtualBox but had some conflicts with that and Docker (*sigh*) on the same host.

Thanks for the new release, I'll kick the tires tomorrow :-)

**Comment by clach04** [comment]
For anyone interested in this, update to fix broken GItHub actions after GitHub change to the environment https://github.com/clach04/pebble_watchface_framework/commit/1ffa11249cb8fd8f12d516f3c826e919d5117ddf
